nodejs/webrtc-hello-video/public/webrtc_mgr.js

```diff
@@ -0,0 +1,521 @@
+const bandwidthRtc = new BandwidthRtc();
+//
+// configuration
+const DEBUG = true;
+const statusDiv = null;
+// a div to append log messages to on screen
+const logDiv = false;
+// a div to append new media elements to (video or audio)
+//  not needed if you implement onNewStream
+const mediaDiv = false;
+const videoEnabled = true;
+// ring while waiting for first connection
+const enableRinging = false;
+// when media connects, mute your mic by default
+const start_muted_audio = false;
+
+//
+//  internal global vars, don't set these
+// universal id for the call (the server tells us this)
+let internal_call_id = "";
+let other_callers = [];
+
+// global vars
+var my_media_stream;
+var my_screen_stream;
+var local_video_stream = false;
+
+/**
+ * Get the token required to auth with the media server
+ * Use that token to start streaming
+ * @param call_info json object with the following
+ *  caller: {name: "adam"}
+ *  call_type: phone", or "push"
+ *  room: room_name
+ *  audio: true OR false (usually true)
+ *  video: true OR false
+ */
+async function getOnline(call_info) {
```

Comment:
This function is specific to this app, so it probably wouldn't go into a library, but instead into the mainline JS for the application (which might reference the helpers)

Reply:
it's *almost* generic, but not quite, I've used variants for 3 apps now. So there's something there. But I haven't figured out how to make it right for more use cases yet.